sonar-scm-git-plugin/src/main/java/org/sonarsource/scm/git/GitScmProvider.java

```diff
@@ -47,4 +67,50 @@ public boolean supports(File baseDir) {
   public BlameCommand blameCommand() {
     return this.jgitBlameCommand;
   }
+
+  @Nullable
+  @Override
+  public Collection<Path> branchChangedFiles(String targetBranchName, Path rootBaseDir) {
+    RepositoryBuilder builder = new RepositoryBuilder().findGitDir(rootBaseDir.toFile());
+    if (builder.getGitDir() == null) {
+      LOG.warn("Not inside a Git work tree: {}", rootBaseDir);
+      return null;
+    }
+
+    try {
+      Repository repo = builder.build();
+
+      Ref targetRef = repo.exactRef("refs/heads/" + targetBranchName);
+      if (targetRef == null) {
+        LOG.warn("Could not find ref: {}", targetBranchName);
+        return null;
+      }
+
+      Git git = new Git(repo);
+      return git.diff().setShowNameAndStatusOnly(true).setOldTree(prepareTreeParser(repo, targetRef)).call().stream()
+        .map(diffEntry -> repo.getWorkTree().toPath().resolve(diffEntry.getNewPath()))
+        .collect(Collectors.toList());
```

Comment:
Isn't a HashSet more appropriate? Excludes repeated files (if that's possible) and makes `contains()` much faster